### ISSUE TITLE
docs: solver comparison example — regular vs staggered grids

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#46]
## Summary

Add an interactive example (Jupyter notebook + rendered docs page) comparing the four same-BC spectral solver families on a **single canonical problem**, highlighting how grid placement (regular vs staggered) changes boundary handling, eigenvalues, and solution accuracy.

This is a foundational tutorial that should be the first thing users read after the API reference.

## Motivation

Users coming from finite-difference or finite-volume backgrounds need to build intuition for **why** DST-I ≠ DST-II and DCT-I ≠ DCT-II. The difference is subtle (vertex-centred vs cell-centred) but critical for choosing the right solver.

## Canonical Problem

Solve the 2D Poisson equation on the unit square `[0, 1]²`:

```
∇²ψ = f(x, y)
```

with a manufactured solution `ψ_exact = sin(2πx) sin(2πy)` so the RHS is known analytically:

```
f(x, y) = -8π² sin(2πx) sin(2πy)
```

## File Locations

| File | Purpose |
|------|---------|
| `docs/examples/solver_comparison.md` | Rendered docs page (mkdocs) |
| `notebooks/solver_comparison.py` | Jupytext percent-format notebook |
| `docs/examples/index.md` | Examples landing page (add link) |
| `mkdocs.yml` | Add nav entry under Examples |

### Section 1: Grid Geometry (ASCII + matplotlib)

Show the four grid layouts on a small 4×4 domain. ASCII sketch for the docs page:

```
Regular (vertex-centred)          Staggered (cell-centred)

  ψ=0 (boundary)                   |  boundary (half-spacing outside)
  |                                 |
  o----o----o----o----o             |--x----x----x----x--|
  |    |    |    |    |             |  |    |    |    |  |
  o    o    o    o    o             |--x    x    x    x--|
  |    |    |    |    |             |  |    |    |    |  |
  o    o    o    o    o             |--x    x    x    x--|
  |    |    |    |    |             |  |    |    |    |  |
  o----o----o----o----o             |--x----x----x----x--|
  |                                 |
  ψ=0 (boundary)                   |  boundary (half-spacing outside)

  o = grid point (vertex)           x = grid point (cell centre)
  N=5 points span [0,1]             N=4 points span [dx/2, 1-dx/2]
  dx = 1/(N-1) = 0.25               dx = 1/N = 0.25
```

**matplotlib figure:** Two subplots showing grid points as dots overlaid on the domain, with boundary locations marked as dashed lines.

### Section 2: Eigenvalue Spectra

Plot the 1D eigenvalues for all four transform types on the same axes:

```
  λ_k
  0 ─────────────────────────── k=0
  │  ╲  DCT-I (Neumann regular)
  │   ╲
  │    ╲ DCT-II (Neumann staggered)
  │     ╲
  │      ╲ DST-II (Dirichlet staggered)
  │       ╲
  │        ╲ DST-I (Dirichlet regular)
  │         ╲
  -4/dx² ────╲───────────────── k=N-1
```

**matplotlib figure:** Line plot of `eigenvalues(k)` for k=0..N-1 for all four types, N=32. Show that:
- DCT-I and DCT-II both have λ_0 = 0 (null mode)
- DST-I and DST-II are all strictly negative
- The curves differ in curvature due to `2(N±1)` vs `2N` denominators

### Section 3: Solver Comparison Table

Solve the manufactured problem at N=32, 64, 128 and report:

```
┌────────────┬──────────┬──────────────┬────────────┐
│ Solver     │ BC Type  │ Grid Type    │ L∞ error   │
├────────────┼──────────┼──────────────┼────────────┤
│ DST-I      │ Dirichlet│ Regular      │ ≈ 1e-5     │
│ DST-II     │ Dirichlet│ Staggered    │ ≈ 1e-5     │
│ DCT-I      │ Neumann  │ Regular      │ ≈ 1e-5     │
│ DCT-II     │ Neumann  │ Staggered    │ ≈ 1e-5     │
│ FFT        │ Periodic │ —            │ ≈ 1e-12    │
└────────────┴──────────┴──────────────┴────────────┘
```

**matplotlib figure:** Log-log convergence plot (N vs L∞ error) showing second-order convergence for DST/DCT solvers and spectral convergence for FFT.

### Section 4: Solution Visualisation

**matplotlib figure:** 2×2 panel of `pcolormesh` plots:

```
┌─────────────────┬─────────────────┐
│ DST-I solution  │ DST-II solution │
│ (regular Dir.)  │ (staggered Dir.)│
├─────────────────┼─────────────────┤
│ DCT-I solution  │ DCT-II solution │
│ (regular Neu.)  │ (staggered Neu.)│
└─────────────────┴─────────────────┘
```

Each panel shows the computed solution ψ with grid points overlaid as small dots.

### Section 5: Error Maps

**matplotlib figure:** Same 2×2 layout but showing `|ψ_computed - ψ_exact|` to highlight where errors concentrate (boundaries vs interior).

## Code Structure (notebook)

```python
# %% [markdown]
# # Solver Comparison: Regular vs Staggered Grids

# %% Grid setup
import jax.numpy as jnp
from spectraldiffx import (
    solve_poisson_dst, solve_poisson_dst2,
    solve_poisson_dct1, solve_poisson_dct,
    solve_poisson_fft,
    dst1_eigenvalues, dst2_eigenvalues,
    dct1_eigenvalues, dct2_eigenvalues,
    fft_eigenvalues,
)

# %% Section 1: Grid geometry
# ... build regular and staggered grids, plot

# %% Section 2: Eigenvalue spectra
# ... compute and plot eigenvalues

# %% Section 3: Manufactured solution + convergence
# ... loop over N, solve, compute errors, plot log-log

# %% Section 4 & 5: Solution + error visualisation
# ... 2x2 panels
```

## Acceptance Criteria

- [ ] Notebook runs end-to-end with `uv run jupyter execute`
- [ ] All five figures render in the docs build
- [ ] Docs page includes both code cells and narrative prose explaining each figure
- [ ] Eigenvalue section explicitly states the formula difference (`2(N+1)` vs `2N` etc.)
- [ ] Convergence plot confirms second-order for DST/DCT, spectral for FFT

## Labels

`documentation`, `examples`
